embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -462,12 +462,18 @@ public InputStream openNext() throws IOException
             if (!iterator.hasNext()) {
                 return null;
             }
-            String key = iterator.next();
-            GetObjectRequest request = new GetObjectRequest(bucket, key);
-            S3Object obj = client.getObject(request);
-            long objectSize = obj.getObjectMetadata().getContentLength();
-            LOGGER.info("Open S3Object with bucket [{}], key [{}], with size [{}]", bucket, key, objectSize);
-            return new ResumableInputStream(obj.getObjectContent(), new S3InputStreamReopener(client, request, objectSize, retryExec));
+            final String key = iterator.next();
+            final GetObjectRequest request = new GetObjectRequest(bucket, key);
+            return new DefaultRetryable<ResumableInputStream>("Opening the file") {
+                @Override
+                public ResumableInputStream call() throws Exception
+                {
+                    S3Object obj = client.getObject(request);
+                    long objectSize = obj.getObjectMetadata().getContentLength();
+                    LOGGER.info("Open S3Object with bucket [{}], key [{}], with size [{}]", bucket, key, objectSize);
+                    return new ResumableInputStream(obj.getObjectContent(), new S3InputStreamReopener(client, request, objectSize, retryExec));
```

Comment:
`return new ResumableInputStream(obj.getObjectContent()...` is wrapped with another retry logic.
Is it ok?

Reply:
I think these are 2 different non-overlapping retrying purposes: `DefaultRetryable` supposes to retry for failures on when obtaining the object metadata with `getObject()` (perhaps weak consistency of list vs read objects, ...). `ResumableInputStream` will retry/reopen a new stream of the same object when there are any failures while reading the bytes off the content stream (network problems, ...).